streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStreamThreadTest.java

```diff
@@ -90,6 +92,13 @@ public String newStoreName(final String prefix) {
                 "store-"
             );
 
+        final ProcessorSupplier<Object, Object, Void, Void> processorSupplier = () ->
+            new ContextualProcessor<Object, Object, Void, Void>() {
+                @Override
+                public void process(final Record<Object, Object> record) {
+                }
```

Comment:
Huh, I'm surprised this works; I would have expected that the processor has to put the record in the store. If the test does still pass this way, it might reveal that the test is actually not evaluating anything.

Reply:
Looking at the tests, they are more related to the thread and store states more than value checking. Should we add some store updates as part of the processor for _more_ correctness?

Reply:
Thanks for the context. While that might be a good idea, let's keep these PRs more strictly scoped. I'm afraid adding in extra stuff will make it harder for people in the future to understand which changes were part of this transition and which weren't.

Offhand, I'd guess that there's another test somewhere verifying that global stores actually function properly. I think you're right: this test is more about verifying that the thread implementation works as desired.